src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -23,12 +23,17 @@ namespace Microsoft.CodeAnalysis.Host
     [ExportWorkspaceServiceFactory(typeof(ITemporaryStorageService), ServiceLayer.Default), Shared]
     internal partial class TemporaryStorageServiceFactory : IWorkspaceServiceFactory
     {
+        private readonly IWorkspaceThreadingService? _workspaceThreadingService;
+
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public TemporaryStorageServiceFactory()
+        public TemporaryStorageServiceFactory(
+            [Import(AllowDefault = true)] IWorkspaceThreadingService? workspaceThreadingService)
         {
+            _workspaceThreadingService = workspaceThreadingService;
         }
 
+        [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
```

Comment:
oh nice.  can we make the analyzer/fixer do this like it does for the constructors?

Reply:
That would be nice, but could be difficult due to cross-file analysis requirements.